### DISCUSSION TITLE
problem with multi connection string and assignRole

### DISCUSSION THREAD

**Discussion by maxdiable** [spatie/laravel-permission#2327] [Q&A]
hi, I followed your directives when you have multi connections, so I created 2 models (role, permission) like this:

<?php

namespace App\Models;

use App\Enums\OracleConn;
use Spatie\Permission\Models\Role as SpatieRole;

class Role extends SpatieRole
{
    protected $connection = OracleConn::CONN_ORACLE_AUTH;
}

---------------------------------------------

<?php

namespace App\Models;

use App\Enums\OracleConn;
use Spatie\Permission\Models\Permission as SpatiePermission;

class Permission extends SpatiePermission
{
    protected $connection = OracleConn::CONN_ORACLE_AUTH;
}
--------------------------------

this way everything works, but when I try to assign a role to a user (Auth::user()->assignRole('admin');) it gives me this error.
From what I understand is that the command does not recognize the connection.

the errors is:

SQLSTATE[HY000] [1045] Access denied for user 'forge'@'localhost' (using password: NO) (SQL: select * from `roles` where (`team_id` is null or `team_id` is null) and `name` = admin and `guard_name` = jwt limit 1)

any help ?

br
Max

**Comment by parallels999** [comment]
That problem is with your DB user permissions, not on this package
on user `forge`, also you did activate teams feature, but no team is setted

**Reply by parallels999** [reply]
>problem with multi connection
>Access denied for user 'forge'@'localhost' (using password: NO) 

user `forge` has no permission on connection, the problem is not PHP

**Comment by maxdiable** [comment]
I have activated team and the creation of both the user, roles and permissions works perfectly the only thing wrong is the **assignRole** function :(
br
Max

**Reply by fredsal** [reply]
What means "br Max"?

**Comment by maxdiable** [comment]
this is my code: 

permmission model:

`<?php

namespace App\Models;

use App\Enums\OracleConn;
use Spatie\Permission\Models\Permission as SpatiePermission;

class Permission extends SpatiePermission
{
    protected $connection = OracleConn::CONN_ORACLE_AUTH;
}
`

-----------

`<?php

namespace App\Models;

use App\Enums\OracleConn;
use Spatie\Permission\Models\Role as SpatieRole;

class Role extends SpatieRole
{
    protected $connection = OracleConn::CONN_ORACLE_AUTH;
}
`

**Reply by parallels999** [reply]
Please learn to use markdown, that is impossible to read, 
also keep the conversation in a single thread, you open a new one for each reply

**Example:**
```php
<?php

namespace App\Models;

use App\Enums\OracleConn;
use Spatie\Permission\Models\Permission as SpatiePermission;

class Permission extends SpatiePermission
{
    protected $connection = OracleConn::CONN_ORACLE_AUTH;
}
```
```php
<?php

namespace App\Models;

use App\Enums\OracleConn;
use Spatie\Permission\Models\Role as SpatieRole;

class Role extends SpatieRole
{
     protected $connection = OracleConn::CONN_ORACLE_AUTH;
}
```

**Comment by maxdiable** [comment]
i am sorry.. :(
I apologize for my inexperience..

**Reply by fredsal** [reply]
> also keep the conversation in a single thread, you open a new one for each reply

This is a new one 😄😄😄

**Comment by maxdiable** [comment]
ok.. i show you my permission model:


``` php
namespace App\Models;

use App\Enums\OracleConn;
use Spatie\Permission\Models\Permission as SpatiePermission;

class Permission extends SpatiePermission
{
    protected $connection = OracleConn::CONN_ORACLE_AUTH;
}
```

**Reply by fredsal** [reply]
> also keep the conversation in a single thread, you open a new one for each reply

Another  new one 😄😄😄

**Comment by maxdiable** [comment]
Best regards

Il lun 13 feb 2023, 18:05 fredsal ***@***.***> ha scritto:

**Comment by erikn69** [comment]
[Access denied for user 'forge'@'localhost' (using password: NO)](https://www.google.com/search?q=Access+denied+for+user+%27forge%27%40%27localhost%27&oq=Access+denied+for+user+%27forge%27%40%27localhost%27)
